implementations/elixir/ockam/ockam/lib/ockam/transport/uds/listener.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Ockam.Transport.UDS.Listener do
+  use Task, restart: :transient
+
+  alias Ockam.Transport.UDS.ConnectionSupervisor
+
+  require Logger
+
+  @default_socket "/tmp/ockam_uds_server.sock"
+
+  def start_link(opts \\ []) do
+    Task.start_link(__MODULE__, :run, [Keyword.get(opts, :path, @default_socket)])
+  end
+
+  def run(path) do
+    case :gen_tcp.listen(0, [
+           :binary,
+           {:active, true},
```

Comment:
If we listen in the active mode, does it mean that it's possible to get connection messages on the wrong process if there is some race condition between accept and change of controlling process?

Reply:
Good point, perhaps one way to deal with this would be to start with passive or active once and switch it once the Connection child is up.